src/cascadia/LocalTests_TerminalApp/TabTests.cpp

```diff
@@ -817,6 +819,159 @@ namespace TerminalAppLocalTests
+        // -------------------
+        // |   1    |   2    |
+        // |        |        |
+        // -------------------
+        // |   3    |   4    |
+        // |        |        |
+        // -------------------
+        auto result = RunOnUIThread([&page]() {
+            // We start with 1 tab, split vertically to get
+            // -------------------
+            // |   1    |   2    |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Vertical, SplitType::Duplicate, 0.5f, nullptr);
+            // After this the `2` pane is focused, go back to `1` being focused
+            page->_MoveFocus(FocusDirection::Left);
+            // Split again to make the 3rd tab
+            // -------------------
+            // |   1    |        |
+            // |        |        |
+            // ---------|   2    |
+            // |   3    |        |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Horizontal, SplitType::Duplicate, 0.5f, nullptr);
+            // After this the `3` pane is focused, go back to `2` being focused
+            page->_MoveFocus(FocusDirection::Right);
+            // Split to create the final pane
+            // -------------------
+            // |   1    |   2    |
+            // |        |        |
+            // -------------------
+            // |   3    |   4    |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Horizontal, SplitType::Duplicate, 0.5f, nullptr);
+
+            VERIFY_ARE_EQUAL(1u, page->_tabs.Size());
+            auto tab = page->_GetTerminalTabImpl(page->_tabs.GetAt(0));
+            VERIFY_ARE_EQUAL(4, tab->GetLeafPaneCount());
+            // Our currently focused pane should be `4`
+            VERIFY_ARE_EQUAL(4u, tab->_activePane->Id().value());
+        });
+        VERIFY_SUCCEEDED(result);
+
+        // Now try to move the pane through the tree
+        Log::Comment(L"Move pane to the left. This should swap panes 3 and 4");
+        // -------------------
+        // |   1    |   2    |
+        // |        |        |
+        // -------------------
+        // |   4    |   3    |
+        // |        |        |
+        // -------------------
+        result = RunOnUIThread([&page]() {
+            // Set up action
+            MovePaneArgs args{ FocusDirection::Left };
+            ActionEventArgs eventArgs{ args };
+
+            page->_HandleMovePane(nullptr, eventArgs);
+            auto tab = page->_GetTerminalTabImpl(page->_tabs.GetAt(0));
+            VERIFY_ARE_EQUAL(4, tab->GetLeafPaneCount());
+            // Our currently focused pane should be `4`
+            VERIFY_ARE_EQUAL(4u, tab->_activePane->Id().value());
+
+            // Inspect the tree to make sure we swapped
+            VERIFY_ARE_EQUAL(4u, tab->_rootPane->_firstChild->_secondChild->Id().value());

```

Comment:
I am having a bit of troubles with this test with (presumably) threading issues. Specifically when running the test, if I attach a debugger and step through everything works correctly, however if I just run it normally it will fail here and the pane's children are the same before and after `_HandleMovePane`. (Yes I know that the IDs are incorrect here right now, I was going to wait to commit again until I figure out getting the test to pass consistently)

Reply:
TBH, all these UI like tests have those sorts of issues. If it were me, I'd just split this up into more `RunOnUIThread` calls, and maybe even insert some `Sleep`s between 😄 

I'd probably split this up.... (see above)

Reply:
I feel dirty adding as many sleeps as I did (they may not all be necessary), but at least the test passes consistently now.